deepmd/pt/modifier/dipole_charge.py

```
@@ -0,0 +1,347 @@
+        Returns
+        -------
+        dict[str, torch.Tensor]
+            Dictionary containing the correction terms:
+            - energy: Energy correction tensor with shape (nframes, 1)
+            - force: Force correction tensor with shape (nframes, natoms+nsel, 3)
+            - virial: Virial correction tensor with shape (nframes, 3, 3)
+        """
+        if box is None:
+            raise RuntimeWarning(
+                "dipole_charge data modifier can only be applied for periodic systems."
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)